pyccel/ast/numpyext.py

```diff
@@ -1311,81 +1317,101 @@ def is_elemental(self):
-    'fabs'      : NumpyFabs,
-    'exp'       : NumpyExp,
-    'log'       : NumpyLog,
-    'sqrt'      : NumpySqrt,
+    'abs'       : PyccelFunctionDef('abs'       , NumpyAbs),
+    'floor'     : PyccelFunctionDef('floor'     , NumpyFloor),
+    'absolute'  : PyccelFunctionDef('absolute'  , NumpyAbs),
+    'fabs'      : PyccelFunctionDef('fabs'      , NumpyFabs),
+    'exp'       : PyccelFunctionDef('exp'       , NumpyExp),
+    'log'       : PyccelFunctionDef('log'       , NumpyLog),
+    'sqrt'      : PyccelFunctionDef('sqrt'      , NumpySqrt),
     # ---
-    'sin'       : NumpySin,
-    'cos'       : NumpyCos,
-    'tan'       : NumpyTan,
-    'arcsin'    : NumpyArcsin,
-    'arccos'    : NumpyArccos,
-    'arctan'    : NumpyArctan,
-    'arctan2'   : NumpyArctan2,
-    # 'hypot'     : NumpyHypot,
-    'sinh'      : NumpySinh,
-    'cosh'      : NumpyCosh,
-    'tanh'      : NumpyTanh,
-    'arcsinh'   : NumpyArcsinh,
-    'arccosh'   : NumpyArccosh,
-    'arctanh'   : NumpyArctanh,
-    # 'deg2rad'   : NumpyDeg2rad,
-    # 'rad2deg'   : NumpyRad2deg,
-    'transpose' : NumpyTranspose,
+    'sin'       : PyccelFunctionDef('sin'       , NumpySin),
+    'cos'       : PyccelFunctionDef('cos'       , NumpyCos),
+    'tan'       : PyccelFunctionDef('tan'       , NumpyTan),
+    'arcsin'    : PyccelFunctionDef('arcsin'    , NumpyArcsin),
+    'arccos'    : PyccelFunctionDef('arccos'    , NumpyArccos),
+    'arctan'    : PyccelFunctionDef('arctan'    , NumpyArctan),
+    'arctan2'   : PyccelFunctionDef('arctan2'   , NumpyArctan2),
+    # 'hypot'     : PyccelFunctionDef('hypot'     , NumpyHypot),
+    'sinh'      : PyccelFunctionDef('sinh'      , NumpySinh),
+    'cosh'      : PyccelFunctionDef('cosh'      , NumpyCosh),
+    'tanh'      : PyccelFunctionDef('tanh'      , NumpyTanh),
+    'arcsinh'   : PyccelFunctionDef('arcsinh'   , NumpyArcsinh),
+    'arccosh'   : PyccelFunctionDef('arccosh'   , NumpyArccosh),
+    'arctanh'   : PyccelFunctionDef('arctanh'   , NumpyArctanh),
+    # 'deg2rad'   : PyccelFunctionDef('deg2rad'   , NumpyDeg2rad),
+    # 'rad2deg'   : PyccelFunctionDef('rad2deg'   , NumpyRad2deg),
+    'transpose' : PyccelFunctionDef('transpose' , NumpyTranspose),
 }
 
-numpy_linalg_functions = {
-    'norm'      : NumpyNorm,
-}
+numpy_mod = Module('numpy',
+    variables = numpy_constants.values(),
+    funcs = numpy_funcs.values(),
+    imports = [
+        Import('linalg', numpy_linalg_mod),
+        Import('random', numpy_random_mod),
+        ])
 
-numpy_random_functions = {
-    'rand'      : NumpyRand,
-    'random'    : NumpyRand,
-    'randint'   : NumpyRandint,
-}
+#==============================================================================
+
+numpy_target_swap = {
+        numpy_funcs['full_like']  : numpy_funcs['full'],
+        numpy_funcs['empty_like'] : numpy_funcs['empty'],
+        numpy_funcs['zeros_like'] : numpy_funcs['zeros'],
+        numpy_funcs['ones_like']  : numpy_funcs['ones']
+    }
```

Comment:
What is this dictionary for?

Reply:
It is for replacing `Import('numpy',AsName(NumpyFullLike,'func'))` with `Import('numpy',AsName(NumpyFull,'func'))`. If I don't swap out the object of the `AsName` in the `Import` target then `_print_Import` prints:
```
from numpy import full_like as full
```

Reply:
Understood! Thanks